cloud/amazon/lambda.py

```diff
@@ -0,0 +1,401 @@
+    description:
+      - The Amazon Resource Name (ARN) of the IAM role that Lambda assumes when it executes your function to access any other Amazon Web Services (AWS) resources
+    default: null
+  handler:
+    description:
+      - The function within your code that Lambda calls to begin execution
+    default: null
+  zip_file:
+    description:
+      - A .zip file containing your deployment package
+    required: false
+    default: null
+    aliases: [ 'src' ]
+  s3_bucket:
+    description:
+      - Amazon S3 bucket name where the .zip file containing your deployment package is stored
+    required: false
+    default: null
+  s3_key:
+    description:
+      - The Amazon S3 object (the deployment package) key name you want to upload
+    required: false
+    default: null
+  s3_object_version:
+    description:
+      - The Amazon S3 object (the deployment package) version you want to upload.
+    required: false
+    default: null
+  description:
+    description:
+      - A short, user-defined function description. Lambda does not use this value. Assign a meaningful description as you see fit.
+    required: false
+    default: null
+  timeout:
+    description:
+      - The function execution time at which Lambda should terminate the function.
+    required: false
+    default: 3
+  memory_size:
+    description:
+      - The amount of memory, in MB, your Lambda function is given
+    required: false
+    default: 128
+notes:
+  - 'Currently this module only supports uploaded code via S3'
+author:
+    - 'Steyn Huizinga (@steynovich)'
+extends_documentation_fragment:
+    - aws
+    - ec2
+'''
+
+EXAMPLES = '''
+# Create Lambda functions
+tasks:
+- name: looped creation
+  lambda:
+    name: '{{ item.name }}'
+    state: present
+    zip_file: '{{ item.zip_file }}'
+    runtime: 'python2.7'
+    role_arn: 'arn:aws:iam::987654321012:role/lambda_basic_execution'
+    handler: 'hello_python.my_handler'
+  with_items:
+    - { name: HelloWorld, zip_file: 'hello-code.zip' }
+    - { name: ByeBye, zip_file: 'bye-code.zip' }
+
+# Basic Lambda function deletion
+tasks:
+- name: Delete Lambda functions HelloWorld and ByeBye
+  lambda:
+    name: '{{ item }}'
+    state: absent
+  with_items:
+    - HelloWorld
+    - ByeBye
+'''
+
+RETURN = '''
+output:
+  description: the data returned by create_function in boto3
+  returned: success
+  type: dict
+  sample:
+    {
+        'FunctionName': 'string',
+        'FunctionArn': 'string',
+        'Runtime': 'nodejs',
+        'Role': 'string',
+        'Handler': 'string',
+        'CodeSize': 123,
+        'Description': 'string',
+        'Timeout': 123,
+        'MemorySize': 123,
+        'LastModified': 'string',
+        'CodeSha256': 'string',
+        'Version': 'string',
+    }
+'''
+
+try:
+    import botocore
+    HAS_BOTOCORE = True
+except ImportError:
+    HAS_BOTOCORE = False
+
+try:
+    import boto3
+    HAS_BOTO3 = True
+except ImportError:
+    HAS_BOTO3 = False
+
+try:
```

Comment:
The `hashlib`, `binascii`, and `base64` conditionals aren't needed - these are in the Python standard library from 2.6 through 3.5, so please remove these conditionals and the checks below (I'll comment there too).